dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py

```diff
@@ -0,0 +1,753 @@
+    all_green_measures_df = all_green_measures_df[
+        all_green_measures_df["INDUSTRY GHG PER UNIT EMISSIONS"] != ":"
+    ]
+
+    # For multiskills the format is different in ENTS - separate them out
+    # ENTS: [[['research'], 'SKILL'], [['This is a cut up', 'a cut up sentence'], 'MULTISKILL']]
+    # to [[['research'], 'SKILL'], [['This is a cut up'], 'SKILL'], [['a cut up sentence'], 'SKILL']]
+
+    def separate_multiskill_ents(entlist):
+        separate_ents = []
+        for ent in entlist:
+            if ent[1] == "MULTISKILL":
+                for sep_ent in ent[0]:
+                    separate_ents.append(
+                        [[sep_ent], "MULTISKILL"]
+                    )  # The format it is expecting
+            else:
+                separate_ents.append(ent)
+        return separate_ents
+
+    all_green_measures_df["ENTS"] = all_green_measures_df["ENTS"].apply(
+        lambda x: separate_multiskill_ents(x) if isinstance(x, list) else x
+    )
+
+    print(f"There are {len(all_green_measures_df)} rows in the merged data")
+    print(f"There are {all_green_measures_df['job_id'].nunique()} unique job ids")
+
+    return all_green_measures_df
+
+
+def add_salaries(all_green_measures_df, job_id_col="job_id"):
+    salary_information = get_mixed_ojo_salaries_sample()
+    salary_information[job_id_col] = salary_information.id.astype(str)
+
+    # add salary
+    all_green_measures_df = pd.merge(
+        all_green_measures_df, salary_information, on=job_id_col, how="left"
+    )
+    return all_green_measures_df
+
+
+def add_locations(all_green_measures_df, job_id_col="job_id"):
+    locations_information = get_mixed_ojo_location_sample()
+    locations_information[job_id_col] = locations_information.id.astype(str)
+
+    # add locations
+    locations_information = locations_information.drop(
+        columns=["is_uk", "is_large_geo", "location", "coordinates"]
+    )
+    all_green_measures_df = pd.merge(
+        all_green_measures_df, locations_information, on=job_id_col, how="left"
+    )
+    return all_green_measures_df
+
+
+def add_sic_info(all_green_measures_df):
+    sic_data = load_sic()
+    sic_names = dict(
+        zip(sic_data["Division"].tolist(), sic_data["Description"].tolist())
+    )
+
+    # Add these new columns
+    all_green_measures_df["SIC_2_digit"] = all_green_measures_df["SIC"].apply(
+        lambda x: str(x)[0:2] if x else None
+    )
+    all_green_measures_df["SIC_2_digit_name"] = all_green_measures_df[
+        "SIC_2_digit"
+    ].apply(lambda x: sic_names.get(x) if x else None)
+
+    return all_green_measures_df
+
+
+def get_soc_info():
+    # we need soc descriptions as well - they have descriptions for 6 digit sic codes
+    soc_descriptions = pd.read_excel(
+        "s3://prinz-green-jobs/inputs/data/occupation_data/ons/extendedsoc2020structureanddescriptionsexcel121023.xlsx",
+        sheet_name="Extended SOC descriptions MG1-9",
+    )
```

Comment:
is there a getter for this? I might be making that up though

Reply:
no - not with the descriptions - I'll add it though since this would be better